be/src/storage/table_reader.cpp

```diff
@@ -14,22 +14,121 @@
 
 #include "storage/table_reader.h"
 
+#include "column/datum_convert.h"
+#include "column/datum_tuple.h"
+#include "storage/chunk_helper.h"
+#include "storage/projection_iterator.h"
+#include "storage/storage_engine.h"
+#include "storage/tablet_manager.h"
+#include "storage/tablet_reader.h"
+#include "storage/union_iterator.h"
+
 namespace starrocks {
 
-TableReader::TableReader(const TableReaderParams& params) : _params(params) {}
+TableReader::TableReader(const TableReaderParams& params) : _params(params) {
+    for (auto& partition : _params.partition_param.partitions) {
+        for (auto& index : partition.indexes) {
```

Comment:
why collect all indexes' tablets? I think we should collect only base index's tablets?

Reply:
the user will ensure there will be one index in each partition_param, and just read the first index